backend/pkg/api/handle_transforms.go

```diff
@@ -0,0 +1,170 @@
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      fmt.Errorf("requester has no permissions to list wasm transforms"),
+				Status:   http.StatusForbidden,
+				Message:  "You don't have permissions to list wasm transforms",
+				IsSilent: false,
+			})
+			return
+		}
+
+		if !api.Cfg.Redpanda.AdminAPI.Enabled {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      fmt.Errorf("you must enable the admin api to manage wasm transforms"),
+				Status:   http.StatusServiceUnavailable,
+				Message:  "you must enable the admin api to manage wasm transforms",
+				IsSilent: false,
+			})
+			return
+		}
+
+		if err := r.ParseMultipartForm(50 * mb); err != nil {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      err,
+				Status:   http.StatusBadRequest,
+				Message:  "Could not parse multipart form",
+				IsSilent: false,
+			})
+			return
+		}
+
+		metadataJSON := r.FormValue("metadata")
+		if metadataJSON == "" {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      fmt.Errorf("metadata field is missing"),
+				Status:   http.StatusBadRequest,
+				Message:  "Could not find or parse form field metadata",
+				IsSilent: false,
+			})
+			return
+		}
+
+		var msg TransformMetadata
+		if err := json.Unmarshal([]byte(metadataJSON), &msg); err != nil {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      err,
+				Status:   http.StatusBadRequest,
+				Message:  "Could not decode transform metadata",
+				IsSilent: false,
+			})
+			return
+		}
+
+		wasmForm, _, err := r.FormFile("wasm_binary")
+		if err != nil {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      err,
+				Status:   http.StatusBadRequest,
+				Message:  "Could not find or parse form field wasm_binary",
+				IsSilent: false,
+			})
+			return
+		}
+		defer wasmForm.Close()
+
+		wasmBinary, err := io.ReadAll(wasmForm)
+		if err != nil {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      err,
+				Status:   http.StatusBadRequest,
+				Message:  "Could not read wasm binary",
+				IsSilent: false,
+			})
+		}
+
+		if err := api.RedpandaSvc.DeployWasmTransform(r.Context(), adminapi.TransformMetadata{
+			Name:         msg.Name,
+			InputTopic:   msg.InputTopic,
+			OutputTopics: msg.OutputTopics,
+			Status:       nil,
+			Environment:  msg.Environment,
+		}, wasmBinary); err != nil {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      err,
+				Status:   http.StatusInternalServerError,
+				Message:  "Could not deploy wasm transform",
+				IsSilent: false,
+			})
+			return
+		}
+
+		transforms, err := api.RedpandaSvc.ListWasmTransforms(r.Context())
+		if err != nil {
+			rest.SendRESTError(w, r, api.Logger, &rest.Error{
+				Err:      err,
+				Status:   http.StatusInternalServerError,
+				Message:  "Could not list wasm transforms from Kafka cluster",
+				IsSilent: false,
+			})
+			return
+		}
+
+		t, err := findTransformByName(transforms, msg.Name)
```

Comment:
huh, i guess it should. dunno why i felt differently earlier 😅

Reply:
oh lol i see now. the issue here is that it is under api rather than connect. so it isn't using the util functions i created for connect. the find function here is actually using == over contains. 

i'll rename it for sanity purposes